pymc/sampling/mcmc.py

```diff
@@ -1232,12 +1301,18 @@ def _iter_sample(
 
             yield diverging
     except KeyboardInterrupt:
+        if isinstance(trace, ZarrChain):
+            trace.record_sampling_state(step=step)
         trace.close()
         raise
     except BaseException:
+        if isinstance(trace, ZarrChain):
+            trace.record_sampling_state(step=step)
         trace.close()
```

Comment:
Why the repeated except instead of just `except (KeyboardInterrupt, BaseExcetion)`?

Reply:
I know it was already present before

Reply:
I don't know, that was the way it was before and I didn't touch it.

Reply:
Feel free to clean it up. Otherwise nobody will :)